### PR TITLE
Add default config path

### LINKED ISSUE #361: Default config path
What do you think about adding default config file name?
It's not convenient to write always path to the file. 
It's just a line [here](https://github.com/maglnet/ComposerRequireChecker/blob/4.3.x/src/ComposerRequireChecker/Cli/CheckCommand.php#L60-L65) and above to check for path existing.
I'd like to have any default path, as the `phpunit` has. Suggest to set `composer-require-checker.json` as default.

### PR DESCRIPTION
**Pull request by xepozz**
Fixes #361

- Added default config path `composer-require-checker.json`
- Added a test about having the file
- Added a test about overriding the default file
- Tests about absence the default file already exist

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -84,13 +84,16 @@ extensions and internal symbols like `true` or `false` being undefined.
 bin/composer-require-checker check --config-file=path/to/config.json /path/to/your/project/composer.json
 ``` 
 
+> By default, it uses `composer-require-checker.json` if the file exists.
```
Comment (Ocramius) [inline README.md]: This is a quoted block: IMO no need for quoting this.

File: src/ComposerRequireChecker/Cli/CheckCommand.php [hunk 2]
```diff
@@ -241,6 +242,9 @@ private function getCheckOptions(InputInterface $input): Options
         if (is_string($fileName) === false) {
             return new Options();
         }
+        if (file_exists($fileName) === false) {
+            return new Options();
+        }
```
Comment (Ocramius) [inline src/ComposerRequireChecker/Cli/CheckCommand.php]: I think it's best for the tool to just crash, when the given file is not found.

This shows an interesting case to be tested:

 * file does not exist
 * default setting says to look in `composer-require-checker.json`

I think this highlights that perhaps `composer-require-checker.json` as a default command value is not sufficient.

A test with `--config-file=not-existing.json` that **throws** should be added here.

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/ComposerRequireChecker/Cli/CheckCommand.php (hunk 2 above)
Comment (Ocramius) [reply src/ComposerRequireChecker/Cli/CheckCommand.php]: Something like `if (! file_exists(...) && $file === self::DEFAULT_FILE) {` would suffice (with the testing endorsed above)

Comment (xepozz) [reply src/ComposerRequireChecker/Cli/CheckCommand.php]: Done

**Comment by Ocramius** [comment]
Check the CI failures too, please 😁

File: src/ComposerRequireChecker/Cli/CheckCommand.php [hunk 3]
```diff
@@ -241,6 +243,16 @@ private function getCheckOptions(InputInterface $input): Options
         if (is_string($fileName) === false) {
             return new Options();
         }
+        if (file_exists($fileName) === false) {
+            if ($fileName === self::DEFAULT_CONFIG_PATH) {
+                return new Options();
+            } else {
```
Comment (Ocramius) [inline src/ComposerRequireChecker/Cli/CheckCommand.php]: This `else` can be removed

**Comment by xepozz** [comment]
Hope they will be green

**Review by Ocramius — APPROVED** [review]
LGTM

**Comment by Ocramius** [comment]
Thanks @xepozz!

**Pull request merged**